spring-cloud-gcp-pubsub/pom.xml

```diff
@@ -42,5 +42,34 @@
 			<artifactId>spring-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<!-- SI channel adapters -->
+		<dependency>
+			<groupId>org.springframework.integration</groupId>
+			<artifactId>spring-integration-core</artifactId>
```

Comment:
Well, we usually don't provide nested modules and I keen like we can keep out of that here as well.

My idea is that standard Spring API over the target protocol may really be severed by the technology - storage, pubsub, spanner. But SI Channel Adapters, `LockRegistry`, `MetadataStore`, Leader Election etc., can really be combined in the separate `-integration` module. That can be considered as an optional addition if you would like to get a gain of Spring Integration in your application.

If you decide to go `integration` sub-package, you should be figure out in the future where to place those mentioned components implementations.

Hope I am clear